### ISSUE TITLE
Cartesian product with an empty set should be an empty set

### ISSUE THREAD

**Issue by GrosSacASac** [trekhleb/javascript-algorithms#339]
https://github.com/trekhleb/javascript-algorithms/blob/master/src/algorithms/sets/cartesian-product/cartesianProduct.js#L8

I think it makes more sense to return an empty array
```
// invalid input return null or throw an error
if (!setA || !setB) {
    return null;
}
if (!setA.length || !setB.length) {
    return []; 
}
// or do nothing since the for loop will not trigger anyway
```

**Comment by mik-laj** [comment]
Can you create a PR that will introduce this change?

**Comment by GrosSacASac** [comment]
https://github.com/trekhleb/javascript-algorithms/pull/363

**Comment by GrosSacASac** [comment]
could you have a look ?

**Comment by Veeloxfire** [comment]
I've fixed this on a local repos but currently struggling to make pull requests due to lint errors

**Comment by blondieblue-eyes** [comment]
I need to find out how to unlink or unhack every device help

**Comment by lazarljubenovic** [comment]
You're lost, bud.

**Comment by blondieblue-eyes** [comment]
very t.y.

**Comment by blondieblue-eyes** [comment]
like I need to start somewhere I think maybe start with how to understand
the html?

**Comment by blondieblue-eyes** [comment]
yes I absolutely am. I need to start learning from almost the beginning
where should I start I understand some stuff. but its alot of not English
to me  but I really want to learn.! its of alot of interest to me!
